### PR TITLE
Hummingbird integration with lgbm converter

### LINKED ISSUE #402: Conversion from caffe to onnx through coreml produce paddings with -3.4e+38 value.
Hello, 

I try to use conversion of caffe models to onnx using the path that is documented here, converting to coreml and then converting coreml model to onnx.

In the onnx file for some models, there are padding layers that are created for which padding value is set to constant with value -3.4e+38. This padding value is visible while examining the onnx file with netron, while I do not see anything similar in the coreml model. That the reason why I think this is produced during the onnx conversion, but may be I am wrong.

I don't know if this is a bug, a limitation or a something I am doing wrong. 

The network I used is the Alexnet network provided in DIGITS, with very few modifications. Difficult to attach because this network has a lot of parameters.

I have discovered this error because errors are generated when I try using it with TensorRT, which do not accept non-null values in padding. 

May-be there is a way to force this padding to 0, but I do not find how to do it.

My environment Ubuntu 18.04 in an lxc container, python 3.6.9, caffe model produced with the last BLVC release, coremltools 3.4, onnxmltools 1.7.0.

I tried to generate with opset= 7 and opset = 10.

### LINKED ISSUE #403: LightGBM converter integration with Hummingbird
This issue covers the integration of [Hummingbird](https://github.com/microsoft/hummingbird) with the LightGBM convert. With Hummingbird version 0.0.3 we have brought initial support for [ONNX models] (https://github.com/microsoft/hummingbird/pull/142) (generated by the LightGBM convert). The integration will allow users of the LightGBM convert to have ONNX graphs without any ONNXML operator, meaning the model can be run over hardware accelerators. 

Hummingbird runs over python =>3.5 and for Win, Mac and Linux. We have already tested our ONNX support with the [LightGBM convert tests](https://github.com/microsoft/hummingbird/blob/master/tests/test_onnxml_converter_lightgbm.py) so there should be no technical blocker. @ksaur

### PR DESCRIPTION
**Pull request by interesaaat**
Closes #403.

### CODE REVIEW

**Comment by interesaaat** [comment]
All tests pass. I don't know what is going on with LGTM, the logs look ok. @xadupre @wenbingl

**Comment by fgsilva** [comment]
@interesaaat is there an update regarding this issue?

**Comment by interesaaat** [comment]
@fgsilva the PR is ready, I am waiting for review.

File: .azure-pipelines/win32-conda-CI.yml [hunk 1]
```diff
@@ -68,6 +68,7 @@ jobs:
       echo Test numpy installation... && python -c "import numpy"
       python -m pip install %COREML_PATH% %ONNX_PATH%
       python -m pip install tensorflow-cpu==1.15.0
+      python -m pip install torch==1.5.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
```
[the remark this answers is not in the document]
Comment (interesaaat) [reply .azure-pipelines/win32-conda-CI.yml:71]: Agreed! Pytorch dependency is actually in Hummingbird requirements, however [we found](https://github.com/microsoft/hummingbird/issues/166#issue-644317532) that pytorch is not update on pip for windows. Another option is to install hummingbird directly from source rather than from pip. That would work. Should I install Hummingbird from github?

File: onnxmltools/convert/lightgbm/convert.py [hunk 2]
```diff
@@ -17,7 +18,7 @@
 
 def convert(model, name=None, initial_types=None, doc_string='', target_opset=None,
             targeted_onnx=onnx.__version__, custom_conversion_functions=None,
-            custom_shape_calculators=None):
+            custom_shape_calculators=None, onnx_operators_only=False):
```
[the remark this answers is not in the document]
Comment (interesaaat) [reply onnxmltools/convert/lightgbm/convert.py:21]: Yes! Let me rename it.

File: .azure-pipelines/win32-conda-CI.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (wenbingl) [reply .azure-pipelines/win32-conda-CI.yml:71]: In that case, can we put this line in the hummingbird requirements.txt : torch==1.5.0+cpu -f https://download.pytorch.org/whl/torch_stable.html, 
Or add some lines in the code to log this issue or avoid the confusion?

File: onnxmltools/convert/common/utils.py [hunk 3]
```diff
@@ -5,3 +5,15 @@
 # --------------------------------------------------------------------------
 
 from onnxconverter_common.utils import *  # noqa
+
+
+def hummingbird_installed():
```
Comment (wenbingl) [inline onnxmltools/convert/common/utils.py:10]: please move this func into https://github.com/microsoft/onnxconverter-common/blob/master/onnxconverter_common/utils.py.
 
this file will be deprecated.

File: .azure-pipelines/win32-conda-CI.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (interesaaat) [reply .azure-pipelines/win32-conda-CI.yml:71]: This is complicated because `torch==1.5.0+cpu -f https://download.pytorch.org/whl/torch_stable.html` will not work for linux and mac. We have [this](https://github.com/microsoft/hummingbird/blob/40d8cc9477fc598081b973217e1315c9a0a890c7/setup.py#L25) logic to install PyTorch, but [we are having problems with pip from windows](https://github.com/microsoft/hummingbird/issues/166). I think that installing directly from github is probably the best solution right now.

[the remark this answers is not in the document]
Comment (interesaaat) [reply .azure-pipelines/win32-conda-CI.yml]: Ok installing from github does not work for python 3.7 because of [this](https://dev.azure.com/onnxmltools/onnxmltools/_build/results?buildId=13929&view=logs&j=14d29e63-7f60-55a7-fdd9-83f0bcb9c249&t=d805671f-b036-5223-a743-bcd6fb46af56). Basically `onnx` uses `typing` which for some reason creates problem with Hummingbird. Since you are already manually installing tensorflor, is it a problem if for windows we also manually install pytorch in the pipeline?

**Comment by interesaaat** [comment]
@wenbingl I think I addressed all of your comments. I have only modified the CI files to point to my fork of onnxconverter-common until the [PR](https://github.com/microsoft/onnxconverter-common/pull/134) is not in. I can remove those changes once merged.

**Comment by wenbingl** [comment]
> @wenbingl I think I addressed all of your comments. I have only modified the CI files to point to my fork of onnxconverter-common until the [PR](https://github.com/microsoft/onnxconverter-common/pull/134) is not in. I can remove those changes once merged.

I merged the PR in onnxconverter-common so now you can update your final version and I will merge this PR then.

**Comment by interesaaat** [comment]
Thanks! I am working on it right now. Will push in a minute.
